### PR TITLE
Check local images have expected version

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR checks if local images match the version running in the CI.
It also adds a new option to the `update-resources` task to selectively download images.

### CODE REVIEW

File: tasks/kmt.py [hunk 1]
```diff
@@ -812,6 +812,50 @@ def kmt_sysprobe_prepare(
...
+            if vmid in d.name:
+                check_tag = True
+                break
+
+        if not check_tag:
+            continue
+
+        manifest_file = '.'.join(platinfo["image"].split('.')[:-2]) + ".manifest"
+
+        if not os.path.exists(f"{kmt_os.rootfs_dir / manifest_file}"):
```
Comment (gjulianm) [inline tasks/kmt.py:836]:
```suggestion
        if not (kmt_os.rootfs_dir / manifest_file).exists():
```

File: tasks/kmt.py [hunk 2]
```diff
@@ -812,6 +812,50 @@ def kmt_sysprobe_prepare(
...
+            not_matches.append(platinfo["image"])
+            continue
+
+        with open(f"{kmt_os.rootfs_dir / manifest_file}") as f:
```
Comment (gjulianm) [inline tasks/kmt.py:840]:
```suggestion
        with open(kmt_os.rootfs_dir / manifest_file) as f:
```

File: tasks/kmt.py [hunk 3]
```diff
@@ -812,6 +812,50 @@ def kmt_sysprobe_prepare(
...
+            for line in f:
+                key, value = line.strip().split('=',1)
+                if key != "COMMIT":
```
Comment (gjulianm) [inline tasks/kmt.py:843]: I could add the version to the manifest, that way we would not have to compare with the commit. The main problem with this is that branch images will not match.

Comment (usamasaqib) [reply tasks/kmt.py:843]: Hmm we could add a flag to skip this check for when we want to test branch images. Alternatively you could add the version to the manifest directly.

Comment (gjulianm) [reply tasks/kmt.py:843]: ami-builder PR is merged :D Only issue is watching out here for manifests without `IMAGE_VERSION`

Comment (usamasaqib) [reply tasks/kmt.py:843]: How would that happen?

Comment (gjulianm) [reply tasks/kmt.py:843]: It wouldn't because you're adding here the manifest files so no one will have outdated manifests. My mistake, sorry :P

**Review by gjulianm — APPROVED** [review]

**Pull request merged**